### PR TITLE
Adding final state of pesto app.

### PR DESCRIPTION
**Pull request by jwill**
Adds source code for Pesto Flutter app to be accompanied by codelab.

### CODE REVIEW

File: pesto_flutter/final/LICENSE [hunk 1]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (domesticmouse) [inline pesto_flutter/final/LICENSE:1]: Is this the first release of this code? If it is the first release I'd prefer it under the traditional flutter license used by most of the codelabs in this repo

**Review by domesticmouse — CHANGES_REQUESTED** [review]

File: pesto_flutter/final/LICENSE (hunk 1 above)
Comment (jwill) [reply pesto_flutter/final/LICENSE]: Technically the 2nd release, first was a not-widely circulated release on material-foundation. Will remove local license in favor of top level one.

File: pesto_flutter/final/lib/presentation/app.dart [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/app.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/router.dart [hunk 3]
```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/router.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/screens/notes.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/screens/notes.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/screens/profile.dart [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/screens/profile.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/screens/recipe.dart [hunk 6]
```diff
@@ -0,0 +1,286 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/screens/recipe.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/screens/recipes.dart [hunk 7]
```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/screens/recipes.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/screens/root.dart [hunk 8]
```diff
@@ -0,0 +1,137 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/screens/root.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/screens/settings.dart [hunk 9]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/screens/settings.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/theme.dart [hunk 10]
```diff
@@ -0,0 +1,175 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/theme.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/widgets/adaptive_cards.dart [hunk 11]
```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/widgets/adaptive_cards.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/widgets/filter_list.dart [hunk 12]
```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/widgets/filter_list.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/widgets/image_theme.dart [hunk 13]
```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/widgets/image_theme.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/lib/presentation/widgets/selected_chip.dart [hunk 14]
```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/lib/presentation/widgets/selected_chip.dart:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/linux/main.cc [hunk 15]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/linux/main.cc:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/linux/my_application.cc [hunk 16]
```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/linux/my_application.cc:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/linux/my_application.h [hunk 17]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/linux/my_application.h:4]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

File: pesto_flutter/final/macos/Runner/AppDelegate.swift [hunk 18]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (domesticmouse) [inline pesto_flutter/final/macos/Runner/AppDelegate.swift:3]: Replace with: https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

**Review by domesticmouse — CHANGES_REQUESTED** [review]
To re-license this code requires a fair amount of additional work to change over license headers. Given the code has been released already it's probably best not to re-license at this point. I'd suggest moving the Apache license block to the `pesto_flutter` base directory, as the contents of `final` are going to be replicated about 15 or so times. 

We probably don't need a contribution guide or code of conduct in the `final/docs` directory.

There are a set of images in `final/img` that appear to be supporting material for the codelab itself, not assets for the running application?

There are license headers applied to code that was generated by the template engine, which is going to make maintenance via a `codelab_rebuild` script more complicated going forward, as well.

**Comment by domesticmouse** [comment]
Actually, I'll land this as is and make the changes I need as I build out the steps. Thanks @jwill!

**Review by domesticmouse — APPROVED** [review]

**Pull request merged**